### DISCUSSION TITLE
Update Boxed Row

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1546] [Proposals]
### Goal

Avaliar a proposta de evoluir, corrigir e ajustar pontos específicos no componente Boxed Row.

### Context

Na nova Tab de Casa Inteligente, temos a necessidade de ajustar e evoluir algumas propriedades do componente, que são: 

- Subtitle (true/false)
- Description (true/false)
- Double interaction - Slot (Icon Button)
- Custom Color (ref: Poster Card) 

Para o slot, vamos aplicar o Icon Button, com isso, precisamos confirmar se existe a possibilidade de criar states para o componente, pois no Figma, não está disponível.

- Default
- Selected
- Disabled
- Loading

Encontramos algumas questões no componente que precisam ser corrigidas, segue abaixo.

**BUG Double interaction**

Ao selecionar a opção para aplicar um slot, o Divider que separa uma ação da outra, desaparece.

**BUG Theme Variant/Context**

Ao alterar de ”default” para ”inverse”, o componente perde 2px na altura.

### Impact

Fazendo essas correções e aplicando as evoluções propostas, não precisaríamos criar uma nova versão do Boxed Row na biblioteca Vivo Extended.

### Discussion Questions

N/D

### Do you have a visual proposal?

https://www.figma.com/file/nfPtfLecvO1oy4efGCSUDP/Evolu%C3%A7%C3%A3o-Boxed-Row?type=design&node-id=1%3A86&mode=design&t=jglWX0aTaxkjJeo2-1

**Comment by yceballost** [comment]
It makes sense to be able to enable double interaction when you want to have actions in the right slot. We will add it to both the RowList and the BoxedList. Thanks @annefranca!!

**Reply by annefranca** [reply]
@yceballost e sobre os outras questões solicitadas acima?

Props Boxed Row:
Subtitle (true/false)
Description (true/false)
Custom Color (ref: Poster Card)

State Icon Button:
Default
Selected
Disabled
Loading

**Reply by yceballost** [reply]
I prefer talk about the iconButton in a different discussion, they are different components and it is better not to mix them.

About the "Fila de encajones de accesorios:" we can include these controls in Figma, yes

**Reply by annefranca** [reply]
@yceballost quando essas melhorias estariam disponíveis?

**Reply by yceballost** [reply]
The same message as iconButton discussion, before that we have a lot of things, please, prioritize with your team the "backlog" column in Novo App project in jira

**Reply by AnaMontes11** [reply]
Hi @annefranca 
We would like to share the following regarding some of the points discussed here:
- We are working on defining the rows to include this double interaction. This task is related to defining the new icon button, which we also introduced in yesterday's weekly. This new component is being worked on as a separate task.
- The loss of the 2px in height in the inverse is not a BUG because in the inverse, the border disappears. The component in Figma corresponds to the existing implementation in development.
- Regarding the custom color for rows, we believe this is a critical component that is widely used. Unlike cards, we feel that rows should remain a more "simple" component. Opening up this possibility for them could generate many problems.
- Enabling the title and description as true/false creates breaking changes in Figma, and we need to thoroughly study a solution that minimizes this impact, so we are postponing it for the next Major (13.0.0)

**Reply by annefranca** [reply]
Obrigada, @AnaMontes11 

A solicitação da evolução da propriedade custom color é a principal para aplicarmos o Boxed Row na jornada de Casa Inteligente, caso não for realmente possível, vamos precisar criar um componente Boxed Row Extended.
